docs/src/cookbooks/verilog-vs-chisel.md

```diff
@@ -0,0 +1,920 @@
+
+```scala mdoc:invisible
+import chisel3._
+import chisel3.util.{switch, is}
+import chisel3.stage.ChiselStage
+import chisel3.experimental.ChiselEnum
+import chisel3.util.{Cat, Fill, DecoupledIO}
+```
+
+<body>
+    <!-- This script is needed so that Markdown and HTML will render together, see link to Stack overflow -->
+    <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
+    <table border ="0">
+        <h1>Creating a Module</h1>
+        <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+module Foo (
+  input  a,
+  output b
+);
+  assign b = a;
+endmodule
+```
+
+</td>
+    <td>
+
+```scala mdoc
+class Foo extends Module {
+  val a = IO(Input(Bool()))
+  val b = IO(Output(Bool()))
+  b := a
+}
+```
+</td>
+         </tr>
+    </table>
+</body>
+</html>
+
+# Parameterizing a Module
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+
+<tr>
+<td>
+
+```verilog
+module PassthroughGenerator(
+  input        clock,
+  input        reset,
+  input  [width-1:0] io_in,
+  output [width-1:0] io_out
+);
+ 
+  parameter width = 8;
+  
+  assign io_out = io_in;
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class PassthroughGenerator(width: Int = 8) extends Module {
+  val io = IO(new Bundle {
+    val in = Input(UInt(width.W))
+    val out = Output(UInt(width.W))
+  })
+  io.out := io.in
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new PassthroughGenerator(10))
+```
+</td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+module ParameterizedWidthAdder(
+  input [in0Width-1:0] in0,
+  input [in1Width-1:0] in1,
+  output [sumWidth-1:0] sum
+);
+  parameter in0Width = 8;
+  parameter in1Width = 1;
+  parameter sumWidth = 9;
+
+  assign sum = in0 + in1;
+
+endmodule
+```
+
+</td>
+<td>
+
+```scala mdoc:silent
+class ParameterizedWidthAdder(
+  in0Width: Int,
+  in1Width: Int,
+  sumWidth: Int) extends Module {
+  val io = IO(new Bundle {
+    val in0 = Input(UInt(in0Width.W))
+    val in1 = Input(UInt(in1Width.W))
+    val sum = Output(UInt(sumWidth.W))
+  })
+  // a +& b includes the carry, a + b does not
+  io.sum := io.in0 +& io.in1
+}
+```
+</td>
+</tr>
+<tr>
+<td>
+
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Wire Assignment and Literal Values
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+wire [31:0] 
+a = 32'd42; 
+wire [31:0] 
+b = 32'hbabecafe; 
+wire [15:0] c; 
+assign c = 16'b1;
+```
+
+</td>
+<td>
+
+```scala mdoc:silent
+
+
+class MyWireAssignmentModule extends Module {
```

Comment:
Do we need this example and the following one, or should we just delete this example?